modules/openapi-generator/src/main/resources/cpp-rest-sdk-client/cmake-lists.mustache

```diff
@@ -12,6 +12,8 @@ cmake_minimum_required (VERSION 3.1)
 
 project({{{packageName}}})
 
+option(SHARED "Compile as shared libraries" ON)
```

Comment:
There is no need to introduce a new option. CMake already provides an option for the same purpose:
https://cmake.org/cmake/help/latest/variable/BUILD_SHARED_LIBS.html

Reply:
Hi @srgbtl,
Thanks for your review.
`BUILD_SHARED_LIBS` is a global setting that impacts all the libraries of a project. If the project has several libraries, it does not offer any granularity on which libraries should be static or shared unless forced by hard-coding the value at `add_library` level.
I personally would not recommend to follow this approach. Please let me know if this is a blocker for you, and absolutely requires to be changed.
*(Just for the sake of clarity and understanding why I am answering those comments, I am the tutor of Sanjay).*

Reply:
You can always overwrite this value for the particular project, for example in case add_subdirectory is used - just set BUILD_SHARED_LIBS to TRUE, or apply some patch for generated code. This is probably better than having to set a bunch of nonstandard library-specific variables.

If you really need to make this functionality a part of the generator, please consider making changes in that way which does not break current behavior.

Reply:
Okay, I will update as suggested.

Reply:
@srgbtl I have made changes as suggested. Please review it!